doc/user/flowspec.rst

```diff
@@ -0,0 +1,330 @@
+Network Namespaces. For data traffic to cross VRF boundaries, virtual ethernet interfaces
+are created with private IP adressing scheme.
+
+.. code-block:: frr
+
+   router bgp <ASx>
+    neighbor <A.B.C.D> remote-as <ASz>
+    address-family ipv4 flowspec
+     neighbor A.B.C.D activate
+    exit
+   exit
+   router bgp <ASy> vrf vrf2
+    address-family ipv4 unicast
+     rt redirect import <E.F.G.H:II>
+    exit
+   exit
+
+Flowspec Monitor and troubleshooting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can monitor policy-routing objects by using one of the following commands.
+Those command rely on the filtering contexts configured from BGP, and get the
+statistics information retrieved from the underlying system. In other words, those
+statistics are retrieved from `Linux NetFilter`.
+
+.. index:: show pbr ipset <IPSET> | iptable
+.. clicmd:: show pbr ipset <IPSET> | iptable
+
+About rule contexts, it is possible to know which rule has been configured to
+policy-route some specific traffic. The `show pbr iptable` command displays for
+forwarded traffic, which table is used. Then it is easy to use that table identifier
+to dump the routing table that the forwarded traffic will match.
+
+.. code-block:: frr
+
+   show ip route table <ID>
+
+You can troubleshoot BGP Flowspec, or BGP policy based routing. For instance, if you
+encounter some issues when decoding a Flowspec entry, you should enable `debug bgp
+flowspec`. If you fail to apply the flowspec entry into *zebra*, there should be some
+relationship with policy routing mechanism. Here, `debug bgp pbr [error]` could help.
+
+.. index:: [no] debug bgp flowspec | pbr [error]
+.. clicmd:: [no] debug bgp flowspec | pbr [error]
+
+Ensuring that a Flowspec entry has been correctly installed and that incoming
+traffic is policy-routed correctly can be checked like illustrated below. First
+of all, you must check whether the Flowspec entry has been installed or not.
+
+.. code-block:: frr
+
+   CLI# show bgp ipv4 flowspec 5.5.5.2/32
+    BGP flowspec entry: (flags 0x418)
+      Destination Address 5.5.5.2/32
+      IP Protocol = 17
+      Destination Port >= 50 , <= 90
+      FS:redirect VRF RT:255.255.255.255:255
+      received for 18:41:37
+      installed in PBR (match0x271ce00)
+
+This means that the Flowspec entry has been installed in a `Linux Netfilter iptable`
+named `match0x271ce00`. Once you have confirmation it is installed, you can check
+whether you find the associate entry by executing following command. You can also
+check whether incoming traffic has been matched by looking at counter line.
+
+.. code-block:: frr
+
+   CLI# show pbr ipset match0x271ce00
+   IPset match0x271ce00 type net,port
+        to 5.5.5.0/24:proto 6:80-120 (8)
+           pkts 1000, bytes 1000000
+        to 5.5.5.2:proto 17:50-90 (5)
+           pkts 1692918, bytes 157441374
+
+As you can see, the entry is present. note that a `Linux Netfilter iptable can be used
```

Comment:
Sphinx should be warning you about this unterminated markup.

Reply:
is there a way to get the rendering locally ?